src/NavBar.ts

```diff
@@ -11,21 +11,26 @@ import { getApi } from '@/api/Api'
 export default class NavBar {
     private readonly queryStore: QueryStore
     private readonly mapStore: MapOptionsStore
-    private isIgnoreQueryStoreUpdates = false
+    private ignoreStateUpdates = false
 
     constructor(queryStore: QueryStore, mapStore: MapOptionsStore) {
         this.queryStore = queryStore
-        this.queryStore.register(() => this.onQueryStateChanged())
         this.mapStore = mapStore
-        this.mapStore.register(() => this.onQueryStateChanged())
-        window.addEventListener('popstate', () => this.parseUrlAndReplaceQuery())
+        window.addEventListener('popstate', async () => await this.updateStateFromUrl())
+    }
+
+    async startSyncingUrlWithAppState() {
+        // our first history entry shall be the one that we end up with when the app loads for the first time
+        window.history.replaceState(null, '', this.createUrlFromState())
```

Comment:
For example if we open 'http://localhost:3000' and due to the app state the URL is updated to 'http://localhost:3000/?whatever=here' clicking 'back' should lead us to where we came from, not just 'http://localhost:3000'. That's why we replace the current window history entry here instead of adding a new one.